lib/main.js

```diff
@@ -21,40 +22,55 @@ export default class LinterUI {
   statusBar: StatusBar;
   intentions: Intentions;
   subscriptions: CompositeDisposable;
+  idleCallbacks: Set<*>;
 
   constructor() {
     this.name = 'Linter'
+    this.idleCallbacks = new Set()
     this.signal = new BusySignal()
-    this.treeview = new TreeView()
     this.commands = new Commands()
     this.messages = []
     this.statusBar = new StatusBar()
     this.intentions = new Intentions()
     this.subscriptions = new CompositeDisposable()
 
     this.subscriptions.add(this.signal)
-    this.subscriptions.add(this.treeview)
     this.subscriptions.add(this.commands)
     this.subscriptions.add(this.statusBar)
 
-    this.subscriptions.add(atom.config.observe('linter-ui-default.showPanel', (showPanel) => {
-      if (showPanel && !this.panel) {
-        this.panel = new Panel()
-        this.panel.update(this.messages)
-      } else if (!showPanel && this.panel) {
-        this.panel.dispose()
-        this.panel = null
-      }
-    }))
-    this.subscriptions.add(atom.config.observe('linter-ui-default.showDecorations', (showDecorations) => {
-      if (showDecorations && !this.editors) {
-        this.editors = new Editors()
-        this.editors.update({ added: this.messages, removed: [], messages: this.messages })
-      } else if (!showDecorations && this.editors) {
-        this.editors.dispose()
-        this.editors = null
-      }
-    }))
+    const obsShowPanelCB = window.requestIdleCallback(function observeShowPanel() {
+      this.subscriptions.add(atom.config.observe('linter-ui-default.showPanel', (showPanel) => {
+        this.idleCallbacks.delete(obsShowPanelCB)
```

Comment:
Shouldn't this be out of the config observer into the idle callback's first line? We don't want this on each config change